apis/v1alpha2/httproute_types.go

```diff
@@ -408,6 +408,27 @@ type HTTPQueryParamMatch struct {
 	Value string `json:"value"`
 }
 
+// HTTPMethod describes how to select a HTTP route by matching the HTTP
+// method as defined by
+// [RFC 7231](https://datatracker.ietf.org/doc/html/rfc7231#section-4) and
+// [RFC 5789](https://datatracker.ietf.org/doc/html/rfc5789#section-2).
+// The value is expected in upper case
+// +kubebuilder:validation:Enum=GET;HEADER;POST;PUT;DELETE;CONNECT;OPTIONS;TRACE;PATCH
+type HTTPMethod string
+
+const (
+	HTTPMethodUndefined HTTPMethod = "NONE"
+	HTTPMethodGet       HTTPMethod = "GET"
+	HTTPMethodHeader    HTTPMethod = "HEADER"
```

Comment:
Is this same as `HEAD` method? I've not heard about a `HEADER` method, it seems confusing since we already have HTTP headers as a concept.

Reply:
Typo on my part.